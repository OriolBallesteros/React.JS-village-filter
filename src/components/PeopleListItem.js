import React from 'react';
import { Link } from 'react-router-dom';
import '../styles/PeopleListItem.scss'

const PeopleListItem = props => (
        <Link to={`/selected/${props.habitant.id}`} className='PeopleListItem'>
            <p>{props.habitant.name}</p>
            <p>{props.habitant.age} years old</p>
        </Link>
);

export default PeopleListItem;