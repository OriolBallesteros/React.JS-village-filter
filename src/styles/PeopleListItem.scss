.PeopleListItem {
   border: 1px solid black;
   margin-bottom: 50px;
   display: flex;
   justify-content: space-around;
   width: 70%;
   box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
   transition: 0.3s;
   background-color: #33ffffbd;

   Link {
    color: red;
    }

    p {
    width: 100px;
    
    }

    &:hover {
        box-shadow: 0 12px 24px 0 rgba(0,0,0,0.4);
        background-color: #009999;
    }
}