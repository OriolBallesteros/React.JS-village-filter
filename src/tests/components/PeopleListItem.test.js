import React from 'react';
import {shallow} from 'enzyme';
import PeopleListItem from '../../components/PeopleListItem';
import {dummyData} from '../dummyData';

it('should render App component correctly', ()=>{
    const wrapper = shallow(<PeopleListItem habitant={dummyData[0]}/>);

    expect(wrapper).toMatchSnapshot();
});
