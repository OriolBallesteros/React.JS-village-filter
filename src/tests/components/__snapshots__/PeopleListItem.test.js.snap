// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App component correctly 1`] = `
<Link
  className="PeopleListItem"
  replace={false}
  to="/selected/0"
>
  <p>
    Tobus Quickwhistle
  </p>
  <p>
    306
     years old
  </p>
</Link>
`;
